realm/realm-library/src/main/java/io/realm/internal/modules/CompositeMediator.java

```diff
@@ -106,6 +106,9 @@ public void insert(Realm realm, RealmModel object, Map<RealmModel, Long> cache)
 
     @Override
     public void insert(Realm realm, Collection<? extends RealmModel> objects) {
+        if (objects.isEmpty()) {
```

Comment:
The mediator is only for internal use, so we could can decide where the invariant is checked :) ... I don't know if it is possible to move the null check to the mediator, but that would also be an option IMO